lib/bank_accounts/savings_account.rb

```diff
@@ -0,0 +1,81 @@
+module BankAccounts
+  class SavingsAccount < Account
+    ##--------------------------------------------------------------------------
+    # CONSTANTS
+
+    # limits
+    MINIMUM_BALANCE = 10
+
+    # fees
+    WITHDRAWAL_FEE = 2.00
+
+
+    ##--------------------------------------------------------------------------
+    # INSTANCE METHODS
+
+
+    # create a new savings account.
+    def initialize(id, initial_balance)
+      if initial_balance < MINIMUM_BALANCE
+        raise ArgumentError.new("You cannot create a savings account with less than a $#{ MINIMUM_BALANCE } initial deposit.")
+      end
+
+      super(id, initial_balance)
+
+      return true
+    end
+
+
+    # add interest to account.
+    def add_interest(rate_percentage)
+      if validate_number(rate_percentage)
+        # convert rate from percentage to decimal.
+        rate_decimal = rate_percentage / 100.0
+
+        # calculate interest.
+        interest = @balance * rate_decimal
+
+        # add interest to balance.
+        update_balance(interest)
+
+        # return interest.
+        return interest
+      else
+        return false
+      end
+    end
+
```

Comment:
Another place to just call `validate_number` at the top as a guard clause and get rid of the whole if/else. There's no way for `validate_number` to return false, so there's no way for this method to return `false` either.